lib/fluent/plugin/in_tail.rb

```diff
@@ -78,22 +77,37 @@ def initialize
     attr_reader :paths
 
     def configure(conf)
+      compat_parameters_convert(conf, :parser)
+      parser_config = conf.elements('parse').first
+      if parser_config
+        (1..Fluent::Plugin::MultilineParser::FORMAT_MAX_NUM).each do |n|
+          parser_config["format#{n}"] = conf["format#{n}"] if conf["format#{n}"]
+        end
+      end
+
       super
 
+      unless parser_config
```

Comment:
Why this check is here?
This check should be moved to before `if parser_config` at line 82?